src/nlp/oracle_text_parser.py

```diff
@@ -7,27 +10,75 @@ def get_cost_and_effect_from_action(action):
 
     @staticmethod
     def get_normalized_tokens(raw_string):
-        return raw_string.split(' ')
+        return OracleTextParser.parser(raw_string)
 
     @staticmethod
     def get_action_list_from_raw_card_text(raw_card_text):
         return [action.strip(' ') for action in raw_card_text.split("\n")]
 
     @staticmethod
-    def get_normalized_cost_tokens(raw_cost_string):
+    def get_normalized_cost(raw_cost_string):
         normalized_cost_tokens = OracleTextParser.get_normalized_tokens(raw_cost_string)
         return OracleTextParser.classify_cost_tokens(normalized_cost_tokens)
 
     @staticmethod
     def classify_cost_tokens(cost_tokens):
+        red_cost = 0
+        blue_cost = 0
+        black_cost = 0
+        green_cost = 0
+        white_cost = 0
+        colorless_cost = 0
+        generic_cost = 0
+        life_cost = False
+        discard_cost = False
+        loyalty_cost = False
+        sacrifice_cost = False
+        hybrid_cost = False
+        tap_cost = False
+        untap_cost = False
+
+        for token in cost_tokens:
+            print(token.text, token.lemma_, token.pos_, token.tag_, token.dep_, token.shape_, token.is_alpha, token.is_stop)
+            token = token.text.lower()
+            red_cost += token.count('{r}')
+            blue_cost += token.count('{u}')
+            black_cost += token.count('{b}')
+            green_cost += token.count('{g}')
+            white_cost += token.count('{w}')
+            colorless_cost += token.count('{w}')
+            generic_cost += token.count('{10}') * 10 + \
```

Comment:
for i in range(10):c
    generic_cost += token.count(f'{i+1}') * (i+1)
// Synctatic sugar